crates/bevy_gltf/src/loader.rs

```diff
@@ -734,10 +741,19 @@ fn load_node(
     node_index_to_entity_map: &mut HashMap<usize, Entity>,
     entity_to_skin_index_map: &mut HashMap<Entity, usize>,
     active_camera_found: &mut bool,
+    parent_transform: &Transform,
 ) -> Result<(), GltfError> {
     let transform = gltf_node.transform();
     let mut gltf_error = None;
     let transform = Transform::from_matrix(Mat4::from_cols_array_2d(&transform.matrix()));
+    let world_transform = *parent_transform * transform;
```

Comment:
`Transform * Transform` is a lossy operation. I'm concerned the determinant might end up incorrect in some transform hierarchies.
I suggest passing a `GlobalTransform` around instead to be certain.

Reply:
i'm happy to do this. it'll be slower, but this isn't a hot code path.

in the [gltf spec](https://registry.khronos.org/glTF/specs/2.0/glTF-2.0.html#transformations) it says that you can either use a translation + scale + rotation, or a matrix which must decompose into those parts, and explicitly cannot skew or shear. 

since we only care about the scale, and the scales are just Vec3-multiplied at each step, i'm pretty sure this is safe. i guess i could pass the scale down as a Vec3 instead of the whole transform if that would be clearer / avoid potential future misuse.

Reply:
> in the [gltf spec](https://registry.khronos.org/glTF/specs/2.0/glTF-2.0.html#transformations) it says that you can either use a translation + scale + rotation, or a matrix which must decompose into those parts, and explicitly cannot skew or shear.

I think that implementation note is referring purely to the local transform stored in the gltf format, but the note is unfortunately not making that very clear.

> since we only care about the scale, and the scales are just Vec3-multiplied at each step, i'm pretty sure this is safe.

I don't have the math intuition to know whether the loss of skew and shear still keeps the determinant the same 🤷

Reply:
> I think that implementation note is referring purely to the local transform stored in the gltf format, but the note is unfortunately not making that very clear.

yes, but that's all we're dealing with here. at this point the gltf isn't part of the world heirarchy, we just initialize the transform to IDENTITY at the gltf root node.